Releasenotes inter app (#3170)
SkylineCommunications/dataminer-docs: release-notes/Code_Libraries/Skyline.DataMiner.Core/Skyline.DataMiner.Core.InterAppCalls/Skyline_DataMiner_Core_InterAppCalls_Range_1.0.md

---

# Skyline DataMiner Core InterAppCalls Range 1.0

> [!NOTE]
> Range 1.0.x.x is supported as from **DataMiner 10.1.0**. This is a continuation and split from the now obsolete [Class Library](xref:ClassLibrary_Range_1.2).

### 1.0.1.1

#### Breaking change - Replying to an InterApp message

Where before you were allowed to reply to an InterApp call in several ways that involved setting parameter 9000001 (or a similar PID) in some way, this has now changed to only allow a *.Reply* call on the incoming message.

This change allows our library to dynamically select a more efficient underlying communication protocol if you have a high enough DataMiner version, or to continue using SLNet Subscriptions as a fallback.

>[!NOTE]
> The DIS Validator should help report possible invalid code when updating to the new InterApp branch.

#### Breaking change - isLegacyDestination removed

The *Send* method has removed the *isLegacyDestination* argument. This argument was intended to briefly assist in the move from the obsolete *SLC.Lib.* namespaces to the current library. This change occurred several years ago, and as such, this argument was dropped to improve library code maintainability.

> #### New feature - Enhanced efficiency on InterApp reply calls

Reviewer: should we mention limited reply message size?

Author: Perhaps yes, I'll add it

---
uid: Skyline_DataMiner_Core_InterAppCalls_Range_1.0
---

# Skyline DataMiner Core InterAppCalls Range 1.0

> [!NOTE]

> > Range 1.0.x.x is supported as from **DataMiner 10.1.0**. This is a continuation and split from the now obsolete [Class Library](xref:ClassLibrary_Range_1.2).

Reviewer 2: ```suggestion
> Range 1.0.x.x is supported as from **DataMiner 10.1.0**. This is based on the now obsolete [Class Library](xref:ClassLibrary_Range_1.2).
```

### 1.0.1.1

#### Breaking change - Replying to an InterApp message

Previously, it was possible to reply to an InterApp call in several ways that involved setting parameter 9000001 (or a similar PID) in some way. Now only a *.Reply* call on the incoming message is allowed. This change allows the library to dynamically select a more efficient underlying communication protocol if a high enough DataMiner version is used or to continue using SLNet subscriptions as a fallback.

> [!NOTE]
> The DIS Validator should help report possible invalid code when you update to the new InterApp branch.

#### Breaking change - isLegacyDestination removed

> The *Send* method has removed the *isLegacyDestination* argument. This argument was intended to briefly assist in the move from the obsolete *SLC.Lib.* NuGet to the current library. This change occurred several years ago, and as such, this argument was dropped to improve library code maintainability.

Reviewer 2: ```suggestion
The *isLegacyDestination* argument is now no longer available for the *Send* method. This argument was intended to briefly assist in the move from the obsolete *SLC.Lib.* NuGet to the current library. This change occurred several years ago, and as such, this argument was dropped to improve library code maintainability.
```
